Mail-Merge/Replace-merge-field-with-chart/.NET-Standard/Replace-merge-field-with-chart/Program.cs

```diff
@@ -0,0 +1,184 @@
+                WParagraph paragraph = args.CurrentMergeField.OwnerParagraph;
+                //Gets the field value.
+                List<object[]> graphDetails = args.FieldValue as List<object[]>;
+                //Creates the chart.
+                WChart chart = CreateChart(paragraph.Document, graphDetails);
+                int indexOfField = paragraph.ChildEntities.IndexOf(args.CurrentMergeField);
+                //Clears the field;
+                args.Text = string.Empty;
+                //Inserts the chart at corresponding field location.
+                paragraph.ChildEntities.Insert(indexOfField, chart);
+
+            }
+        }
+        /// <summary>
+        /// Creates the chart based on the graph data.
+        /// </summary>
+        /// <param name="document"></param>
+        private static WChart CreateChart(WordDocument document, List<object[]> graphDetails)
+        {
+
+            //Create the new chart.
+            WChart chart = new WChart(document);
+            chart.Width = 410;
+            chart.Height = 250;
+            chart.ChartType = OfficeChartType.Column_Clustered;
+            //Assign data.
+            AddChartData(chart, graphDetails);
+            //Set a chart title.
+            chart.ChartTitle = "Sales Report";
+            //Set Datalabels.
+            IOfficeChartSerie serie1 = chart.Series.Add("Highest Sale");
+            //Set the data range of chart series – start row, start column, end row and end column.
+            serie1.Values = chart.ChartData[2, 2, 5, 2];
+            IOfficeChartSerie serie2 = chart.Series.Add("Average Sale");
+            //Set the data range of chart series start row, start column, end row and end column.
+            serie2.Values = chart.ChartData[2, 3, 5, 3];
+            IOfficeChartSerie serie3 = chart.Series.Add("Lowest Sale");
+            //Set the data range of chart series start row, start column, end row and end column.
+            serie3.Values = chart.ChartData[2, 4, 5, 4];
+            //Set the data range of the category axis.
+            chart.PrimaryCategoryAxis.CategoryLabels = chart.ChartData[2, 1, 6, 1];
+            //Set legend.
+            chart.HasLegend = true;
+            chart.Legend.Position = OfficeLegendPosition.Bottom;
+            //Hiding major gridlines
+            chart.PrimaryValueAxis.HasMajorGridLines = false;
+            return chart;
+        }
+
+        /// <summary>
+        /// Set the values for the chart.
+        /// </summary>
+        private static void AddChartData(WChart chart, List<object[]> graphDetails)
+        {
+            //Set the value for chart data.
+            int rowIndex = 1;
+            int colIndex = 1;
+            //Get the value from the DataTable and set the value for chart data
+            foreach (object[] row in graphDetails)
+            {
+                foreach (object val in row)
+                {
+                    object value = val.ToString();
```

Comment:
object value = val.ToString();

This code is not needed. Because already datatype is object.

Use below code snippet

foreach (object value in row)
{
         chart.ChartData.SetValue(rowIndex, colIndex, value);
         colIndex++;
          if (colIndex == 5)
               break;
}